### PR TITLE
Created read endpoint for tasks, results and related data [SCI-2667] [SCI-2669]

### PR DESCRIPTION
**Pull request by Zanz2**

### CODE REVIEW

**Comment by Zanz2** [comment]
This includes the tasks jira task and the results jira task as specified in the title, its easier to create them on the same branch for testing, and because results depends on tasks.

File: app/controllers/api/v1/inventory_columns_controller.rb [hunk 1]
```diff
@@ -6,7 +6,7 @@ class InventoryColumnsController < BaseController
       before_action :load_team
       before_action :load_inventory
       before_action :load_inventory_column, only: %i(show update destroy)
-      before_action :check_manage_permissions, only: %i(create update destroy)
```
Comment (okriuchykhin) [inline app/controllers/api/v1/inventory_columns_controller.rb]: You removed this check, but didn't add anything to create method?

File: app/controllers/api/v1/my_module_tags_controller.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          @experiment
+        )
+      end
+
+      def load_task
+        @my_module = @experiment.my_modules.find(params.require(:task_id))
+        render jsonapi: {}, status: :not_found if @my_module.nil?
+      end
+
+      def load_task_repository_row
```
Comment (okriuchykhin) [inline app/controllers/api/v1/my_module_tags_controller.rb]: looks like a typo, I think it should be load_task_tag

File: app/controllers/api/v1/my_modules_controller.rb [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      def outputs
+        outputs = @my_module.my_modules
+                            .page(params.dig(:page, :number))
+                            .per(params.dig(:page, :size))
+        render jsonapi: outputs, each_serializer: MyModuleSerializer
+      end
+
+      def output
+        output = @my_module.my_modules.find(params.require(:id))
+        render jsonapi: {}, status: :not_found if output.nil?
```
Comment (okriuchykhin) [inline app/controllers/api/v1/my_modules_controller.rb]: There is no need for this check, find at previous line will raise an exception if item not found

File: app/controllers/api/v1/my_modules_controller.rb [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      def inputs
+        inputs = @my_module.my_module_antecessors
+                           .page(params.dig(:page, :number))
+                           .per(params.dig(:page, :size))
+        render jsonapi: inputs, each_serializer: MyModuleSerializer
+      end
+
+      def input
+        input = @my_module.my_module_antecessors.find(params.require(:id))
+        render jsonapi: {}, status: :not_found if input.nil?
```
Comment (okriuchykhin) [inline app/controllers/api/v1/my_modules_controller.rb]: Same here

File: app/controllers/api/v1/my_modules_controller.rb [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        render jsonapi: {}, status: :not_found if @my_module.nil?
+      end
+
+      # Made the method below because its more elegant than changing parameters
+      # in routes file, and here. It exists because when we call input or output
+      # for a task, the "id" that used to be task id is now an id for the output
+      # or input.
+      def load_task_relative
+        @my_module = @experiment.my_modules.find(params.require(:task_id))
+        render jsonapi: {}, status: :not_found if @my_module.nil?
```
Comment (okriuchykhin) [inline app/controllers/api/v1/my_modules_controller.rb]: same here

File: app/controllers/api/v1/user_my_modules_controller.rb [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+      def load_experiment
+        @experiment = @project.experiments.find(params.require(:experiment_id))
+        render jsonapi: {}, status: :forbidden unless can_read_experiment?(
+          @experiment
+        )
+      end
+
+      def load_task
+        @my_module = @experiment.my_modules.find(params.require(:task_id))
+        render jsonapi: {}, status: :not_found if @my_module.nil?
```
Comment (okriuchykhin) [inline app/controllers/api/v1/user_my_modules_controller.rb]: not needed here

File: app/controllers/api/v1/user_my_modules_controller.rb [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+      end
+
+      def load_user_task
+        @user_task = @my_module.user_my_modules.find(
+          params.require(:id)
+        )
+        render jsonapi: {}, status: :not_found if @user_task.nil?
```
Comment (okriuchykhin) [inline app/controllers/api/v1/user_my_modules_controller.rb]: not needed here

File: app/serializers/api/v1/my_module_repository_row_serializer.rb [hunk 8]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class MyModuleRepositoryRowSerializer < ActiveModel::Serializer
+      type :task_inventory_rows
+      attribute :id
```
Comment (okriuchykhin) [inline app/serializers/api/v1/my_module_repository_row_serializer.rb]: id is included by default, no need to specify it as attribute

File: app/controllers/api/v1/inventory_columns_controller.rb (hunk 1 above)
Comment (Zanz2) [reply app/controllers/api/v1/inventory_columns_controller.rb]: yeah, i thought all users should be able to create columns on inventory, my mistake

Comment (Zanz2) [reply app/controllers/api/v1/inventory_columns_controller.rb]: will attempt to fix all of this tommorow

**Comment by Zanz2** [comment]
Fixed according to pull request review

**Pull request merged**